app/(default)/practice/[situationId]/[detailId]/dial/call/hooks/useVoiceDetection.ts

```
@@ -56,89 +90,161 @@ export function useVoiceDetection({
 		analyserNodeRef.current = null;
 		setAnalyserNode(null);
+
 		frameCountRef.current = 0;
 		speechFrameCountRef.current = 0;
 	}, []);
 
-	// 마이크 및 Voice Detection 초기화
 	useEffect(() => {
 		if (!isRecording) {
+			// 녹음이 꺼지면 리소스 정리
 			cleanup();
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

/home/jailuser/git/app/(default)/practice/[situationId]/[detailId]/dial/call/hooks/useVoiceDetection.ts:101:4
   99 | 		if (!isRecording) {
  100 | 			// 녹음이 꺼지면 리소스 정리
> 101 | 			cleanup();
      | 			^^^^^^^ Avoid calling setState() directly within an effect
  102 | 			return;
  103 | 		}
  104 |

(react-hooks/set-state-in-effect)